go/cmd/api-gateway/consume.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Jorrit05/DYNAMOS/pkg/lib"
+	pb "github.com/Jorrit05/DYNAMOS/pkg/proto"
+)
+
+func handleIncomingMessages(ctx context.Context, grpcMsg *pb.SideCarMessage) error {
+	logger.Debug("start api gateway handleIncomingMessages")
+	ctx, span, err := lib.StartRemoteParentSpan(ctx, serviceName+"/func: handleIncomingMessages", grpcMsg.Traces)
+	if err != nil {
+		logger.Sugar().Warnf("Error starting span: %v", err)
+	}
+	defer span.End()
+
+	logger.Sugar().Debugw("Type:", "MessageType", grpcMsg.Type)
+
+	switch grpcMsg.Type {
+	case "requestApprovalResponse":
```

Comment:
the case should mostly match the struct type, which it does not do here:
		var requestApproval pb.RequestApproval

Maybe good to draw a simplified view of the message flow? Than we can discuss message flow more easily

So how would we do it now if the request was not accepted? 

Maybe it is good to change the message type  "RequestApproval" to the one you mentio here:
"requestApprovalResponse" which then might contain an error field

Reply:
That makes sense, now we have the two sides of communication tho, 

so 
(1) Api Gateway -- requestApprovalRequest (RequestApproval) --> Orchestrator 
and then
(2) Orchestrator --- requestApprovalResponse (AcceptedDataRequest) --> Api Gateway

i think changing the RequestApproval might be a bit confusing since it is the initial request, I do however agree that it does not make sense to have AccepetedDataRequest as the response since it might contain errors.

So maybe we should change AcceptedDataRequest to RequestApprovalResponse? and that can include the errors?

what do you think?

Reply:
ah right, I see now that in my comment I messed it up. 

So yeah datatype AcceptedDataRequest should change to -> requestApprovalResponse

And this one requestApprovalRequest, might (let's discuss)  be shortened to simply 'requestApproval' its simply passing it on